clang/lib/CIR/Dialect/Transforms/TargetLowering/LowerFunction.cpp

```diff
@@ -42,6 +42,17 @@ Value buildAddressAtOffset(LowerFunction &LF, Value addr,
   return addr;
 }
 
+Value emitPointerBitcast(Value Src, Type DestTy, LowerFunction &CGF) {
+  auto destPtrTy = PointerType::get(CGF.getRewriter().getContext(), DestTy);
+
+  if (auto load = dyn_cast<LoadOp>(Src.getDefiningOp()))
```

Comment:
Why do you need to special case the load here?

Reply:
I want to bitcast the pointer, not the loaded value. Bitcasting aggregate types is one of the main problems with the current implementation. So, this is a workaround.

Reply:
Can you give some CIR snippet to elaborate? I'm curious, perhaps I can give a few more feedback. Do you thing this can get a bit better if we introduce `CastKind::abi` instead?

Reply:
> Why do you need to special case the load here?

IIRC, this is what replaces the old load (struct to struct) with the coerced load (struct to `i64`). Not having it leaves a useless load op.

@bruteforceboy maybe this method could be better named?

Reply:
oh, I see now, thanks for the clarification @sitio-couto